apps/prairielearn/src/admin_queries/course_instance_active_enrollments.sql

```diff
@@ -1,124 +1,135 @@
 WITH
-  course_instances_initial_selection AS (
-    -- First find any course instances with activity in
-    -- the given time period.
-    SELECT DISTINCT
-      ci.id
-    FROM
-      assessment_instances AS ai
-      JOIN assessments AS a ON (a.id = ai.assessment_id)
-      JOIN course_instances AS ci ON (ci.id = a.course_instance_id)
-      JOIN pl_courses AS c ON (c.id = ci.course_id)
-      JOIN institutions AS i ON (i.id = c.institution_id)
-    WHERE
-      (
-        $institution_short_name = ''
-        OR i.short_name = $institution_short_name
-      )
-      AND ai.modified_at BETWEEN $start_date AND $end_date
-  ),
   course_instance_user_data AS (
-    -- Re-select this to get all course instance data,
-    -- not just that from the search date range.
     SELECT
-      ci.id,
+      ci.id AS course_instance_id,
       u.user_id,
       (i.id = u.institution_id) AS is_institution_user,
-      count(*) AS instance_question_count,
-      count(*) FILTER (
-        WHERE
-          iq.modified_at BETWEEN $start_date AND $end_date
-      ) AS instance_question_within_dates_count,
-      min(iq.created_at) AS start_date,
-      max(iq.modified_at) AS end_date
+      count(*) AS instance_question_count
     FROM
-      course_instances_initial_selection AS ciis
-      JOIN course_instances AS ci ON (ci.id = ciis.id)
-      JOIN assessments AS a ON (a.course_instance_id = ciis.id)
+      institutions AS i
+      JOIN pl_courses AS c ON (c.institution_id = i.id)
+      JOIN course_instances AS ci ON (ci.course_id = c.id)
+      JOIN assessments AS a ON (a.course_instance_id = ci.id)
       JOIN assessment_instances AS ai ON (ai.assessment_id = a.id)
       JOIN instance_questions AS iq ON (iq.assessment_instance_id = ai.id)
       JOIN users AS u ON (u.user_id = iq.authn_user_id)
-      JOIN pl_courses AS c ON (c.id = ci.course_id)
-      JOIN institutions AS i ON (i.id = c.institution_id)
+    WHERE
+      i.short_name = $institution_short_name
+      AND ai.modified_at BETWEEN $start_date AND $end_date
+      AND ai.include_in_statistics
+      AND iq.modified_at BETWEEN $start_date AND $end_date
```

Comment:
We are now using a slightly different approach here. We now only include questions where both the question and the assessment instance have `modified_at` within the date range. This could miss some activity if a student starts an assessment before the date range and then continues working during the date range. Or if they start in the date range but continue working after it.

However, I think in practice these cases won't be very common, and it also doesn't matter if we miss the odd assessment or question because it's only the enrollment counts that we really want from this part of the query.